view/rooms/app.js

```diff
@@ -675,8 +675,8 @@ function confirmUserExit() {
     userId: _user.id,
     roomId: parseInt(ROOM_ID, 10),
   };
-
-  return u.postPromiseWithUrl(GET_EVENTS, events);
+  u.postPromiseWithUrl(GET_EVENTS, events);
+  return undefined;
```

Comment:
Is there another way to handle this? 
Returning undefined doesn't sound so good to me. 
Can this be handled in the caller function?

Reply:
I could remove the return entirely, which will default to undefined, right? based on the documentation from mozilla, the idea is that the function we bind to the `beforeunload` event does not return anything so that the browser won't ask the user before he leaves the page. We don't really need the return here.  

What do you mean by caller function, sorry I didn't understand what you mean?

From what I saw here, we're biding the event  `beforeunload` to this `confirmUserExit` function and handling the data there, before user exits the room, 

Is there another place I can control the flow before this function?

Reply:
Sounds weird but that's Mozilla reference 🤷

Reply:
Maybe removing the return will be less confusing if the caller is not expecting any return

Reply:
@LongweiDeng @cassiodias removed the return as suggested and renamed the function being called from **confirm** to **generateUserExitEvent**, hopefully that makes it clearer to understand what we're doing when the event 'beforeunload' is triggered.